perception/occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_nodelet.cpp

```diff
@@ -349,24 +377,75 @@ void OccupancyGridMapOutlierFilterComponent::onOccupancyGridMapAndPointCloud2(
   }
 }
 
+void OccupancyGridMapOutlierFilterComponent::initializerPointCloud2(
+  const PointCloud2 & input, PointCloud2 & output)
+{
+  output.point_step = input.point_step;
+  output.data.resize(input.data.size());
+}
+
+void OccupancyGridMapOutlierFilterComponent::finalizePointCloud2(
+  const PointCloud2 & input, PointCloud2 & output)
+{
+  output.header = input.header;
+  output.point_step = input.point_step;
+  output.fields = input.fields;
+  output.height = input.height;
+  output.is_bigendian = input.is_bigendian;
+  output.is_dense = input.is_dense;
+  output.width = output.data.size() / output.point_step / output.height;
```

Comment:
Though I know that normally height and point_step would not become 0, I prefer adding some error handling here.

```cpp
  // you can write it with rclcpp error handling
  if (output.height == 0 || output.point_step == 0) {
    std::cerr << "Error: output.height or output.point_step cannot be zero." << std::endl;
    return;
  }
```